### ISSUE TITLE
Integration not marked as Failed when Camel is unable to start

### ISSUE THREAD

**Issue by claudio4j** [apache/camel-k#2291]
Camel-K version: 1.4.0

When the integration is built successfully, but the camel application can't start because of some error, the integration isn't marked as {{Failed}} - operator keeps reconciling it.

```
NAME                                       READY   STATUS             RESTARTS   AGE
azure-queue-to-mongodb-57bcdf7f8f-xtfdm    0/1     CrashLoopBackOff   20         82m
```

The integration fails due to an error in camel:

```
oc logs -f azure-queue-to-mongodb-57bcdf7f8f-xtfdm                                      
exec java -cp ./resources:/etc/camel/application.properties:/etc/camel/conf.d/_configmaps/azure-queue-to-mongodb:/etc/camel/resources:/etc/camel/sources/MyRouteBuilder.java:dependencies/app/camel-k-integration-1.4.0.jar:dependencies/lib/boot/io.quarkus.quarkus-bootstrap-runner-1.13.0.Final.jar:dependencies/lib/boot/io.quarkus.quarkus-development-mode-spi-1.13.0.Final.jar:dependencies/lib/boot/io.smallrye.common.smallrye-common-io-1.5.0.jar:dependencies/lib/boot/org.graalvm.sdk.graal-sdk-21.0.0.jar:dependencies/lib/boot/org.jboss.logging.jboss-logging-3.4.1.Final.jar:dependencies/lib/boot/org.jboss.logmanager.jboss-logmanager-embedded-1.0.9.jar:dependencies/lib/boot/org.wildfly.common.wildfly-common-1.5.4.Final-format-001.jar:dependencies/lib/main/com.azure.azure-core-1.12.0.jar:dependencies/lib/main/com.azure.azure-core-http-netty-1.7.1.jar:dependencies/lib/main/com.azure.azure-storage-common-12.10.0.jar:dependencies/lib/main/com.azure.azure-storage-queue-12.8.0.jar:dependencies/lib/main/com.fasterxml.jackson.core.jackson-annotations-2.12.1.jar:dependencies/lib/main/com.fasterxml.jackson.core.jackson-core-2.12.1.jar:dependencies/lib/main/com.fasterxml.jackson.core.jackson-databind-2.12.1.jar:dependencies/lib/main/com.fasterxml.jackson.dataformat.jackson-dataformat-xml-2.12.1.jar:dependencies/lib/main/com.fasterxml.jackson.datatype.jackson-datatype-jdk8-2.12.1.jar:dependencies/lib/main/com.fasterxml.jackson.datatype.jackson-datatype-jsr310-2.12.1.jar:dependencies/lib/main/com.fasterxml.jackson.module.jackson-module-jaxb-annotations-2.12.1.jar:dependencies/lib/main/com.fasterxml.jackson.module.jackson-module-parameter-names-2.12.1.jar:dependencies/lib/main/com.fasterxml.woodstox.woodstox-core-6.1.1.jar:dependencies/lib/main/com.sun.activation.jakarta.activation-1.2.1.jar:dependencies/lib/main/com.sun.istack.istack-commons-runtime-3.0.10.jar:dependencies/lib/main/commons-io.commons-io-2.8.0.jar:dependencies/lib/main/io.micrometer.micrometer-core-1.6.5.jar:dependencies/lib/main/io.netty.netty-buffer-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-codec-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-codec-dns-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-codec-http-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-codec-http2-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-codec-socks-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-common-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-handler-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-handler-proxy-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-resolver-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-resolver-dns-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-tcnative-boringssl-static-2.0.30.Final.jar:dependencies/lib/main/io.netty.netty-transport-4.1.49.Final.jar:dependencies/lib/main/io.netty.netty-transport-native-epoll-4.1.49.Final-linux-x86_64.jar:dependencies/lib/main/io.netty.netty-transport-native-kqueue-4.1.49.Final-osx-x86_64.jar:dependencies/lib/main/io.netty.netty-transport-native-unix-common-4.1.49.Final.jar:dependencies/lib/main/io.projectreactor.netty.reactor-netty-0.9.15.RELEASE.jar:dependencies/lib/main/io.projectreactor.reactor-core-3.3.12.RELEASE.jar:dependencies/lib/main/io.quarkus.arc.arc-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-arc-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-core-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-jackson-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-jaxb-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-jaxp-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-micrometer-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-mongodb-client-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-mutiny-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-mutiny-reactive-streams-operators-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-netty-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-security-runtime-spi-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-smallrye-context-propagation-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-vertx-core-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-vertx-http-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.quarkus-vertx-http-dev-console-runtime-spi-1.13.0.Final.jar:dependencies/lib/main/io.quarkus.security.quarkus-security-1.1.4.Final.jar:dependencies/lib/main/io.smallrye.common.smallrye-common-annotation-1.5.0.jar:dependencies/lib/main/io.smallrye.common.smallrye-common-classloader-1.5.0.jar:dependencies/lib/main/io.smallrye.common.smallrye-common-constraint-1.5.0.jar:dependencies/lib/main/io.smallrye.common.smallrye-common-expression-1.5.0.jar:dependencies/lib/main/io.smallrye.common.smallrye-common-function-1.5.0.jar:dependencies/lib/main/io.smallrye.config.smallrye-config-1.11.1.jar:dependencies/lib/main/io.smallrye.config.smallrye-config-common-1.11.1.jar:dependencies/lib/main/io.smallrye.reactive.mutiny-0.14.0.jar:dependencies/lib/main/io.smallrye.reactive.mutiny-reactive-streams-operators-0.14.0.jar:dependencies/lib/main/io.smallrye.reactive.mutiny-smallrye-context-propagation-0.14.0.jar:dependencies/lib/main/io.smallrye.smallrye-context-propagation-1.1.0.jar:dependencies/lib/main/io.smallrye.smallrye-context-propagation-api-1.1.0.jar:dependencies/lib/main/io.smallrye.smallrye-context-propagation-storage-1.1.0.jar:dependencies/lib/main/io.vertx.vertx-auth-common-3.9.5.jar:dependencies/lib/main/io.vertx.vertx-bridge-common-3.9.5.jar:dependencies/lib/main/io.vertx.vertx-core-3.9.5.jar:dependencies/lib/main/io.vertx.vertx-web-3.9.5.jar:dependencies/lib/main/io.vertx.vertx-web-common-3.9.5.jar:dependencies/lib/main/jakarta.activation.jakarta.activation-api-1.2.1.jar:dependencies/lib/main/jakarta.annotation.jakarta.annotation-api-1.3.5.jar:dependencies/lib/main/jakarta.el.jakarta.el-api-3.0.3.jar:dependencies/lib/main/jakarta.enterprise.jakarta.enterprise.cdi-api-2.0.2.jar:dependencies/lib/main/jakarta.inject.jakarta.inject-api-1.0.jar:dependencies/lib/main/jakarta.interceptor.jakarta.interceptor-api-1.2.5.jar:dependencies/lib/main/jakarta.transaction.jakarta.transaction-api-1.3.3.jar:dependencies/lib/main/net.java.dev.jna.jna-5.3.1.jar:dependencies/lib/main/org.apache.camel.camel-api-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-azure-storage-queue-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-base-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-base-engine-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-bean-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-componentdsl-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-core-catalog-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-core-engine-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-core-languages-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-core-model-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-core-processor-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-core-reifier-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-endpointdsl-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-jackson-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-java-joor-dsl-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-main-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-management-api-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-microprofile-config-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-mongodb-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-support-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-tooling-model-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-util-3.9.0.jar:dependencies/lib/main/org.apache.camel.camel-util-json-3.9.0.jar:dependencies/lib/main/org.apache.camel.k.camel-k-core-1.7.0.jar:dependencies/lib/main/org.apache.camel.k.camel-k-core-api-1.7.0.jar:dependencies/lib/main/org.apache.camel.k.camel-k-core-support-1.7.0.jar:dependencies/lib/main/org.apache.camel.k.camel-k-runtime-1.7.0.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-azure-storage-queue-1.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-bean-1.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-core-1.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-java-joor-dsl-1.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-mongodb-1.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-support-azure-core-1.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-support-common-1.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-support-jackson-dataformat-xml-1.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-support-mongodb-1.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-support-reactor-netty-1.8.1.jar:dependencies/lib/main/org.codehaus.woodstox.stax2-api-4.2.jar:dependencies/lib/main/org.eclipse.microprofile.config.microprofile-config-api-1.4.jar:dependencies/lib/main/org.eclipse.microprofile.context-propagation.microprofile-context-propagation-api-1.0.1.jar:dependencies/lib/main/org.eclipse.microprofile.reactive-streams-operators.microprofile-reactive-streams-operators-api-1.0.1.jar:dependencies/lib/main/org.eclipse.microprofile.reactive-streams-operators.microprofile-reactive-streams-operators-core-1.0.1.jar:dependencies/lib/main/org.glassfish.jakarta.json-1.1.6.jar:dependencies/lib/main/org.glassfish.jaxb.jaxb-runtime-2.3.3-b02.jar:dependencies/lib/main/org.glassfish.jaxb.txw2-2.3.3-b02.jar:dependencies/lib/main/org.hdrhistogram.HdrHistogram-2.1.12.jar:dependencies/lib/main/org.jboss.logging.jboss-logging-annotations-2.2.0.Final.jar:dependencies/lib/main/org.jboss.slf4j.slf4j-jboss-logmanager-1.1.0.Final.jar:dependencies/lib/main/org.jboss.spec.javax.xml.bind.jboss-jaxb-api_2.3_spec-2.0.0.Final.jar:dependencies/lib/main/org.jboss.threads.jboss-threads-3.2.0.Final.jar:dependencies/lib/main/org.jooq.joor-0.9.13.jar:dependencies/lib/main/org.latencyutils.LatencyUtils-2.0.3.jar:dependencies/lib/main/org.mongodb.bson-4.1.0.jar:dependencies/lib/main/org.mongodb.mongodb-crypt-1.0.1.jar:dependencies/lib/main/org.mongodb.mongodb-driver-core-4.1.0.jar:dependencies/lib/main/org.mongodb.mongodb-driver-legacy-4.1.0.jar:dependencies/lib/main/org.mongodb.mongodb-driver-reactivestreams-4.1.0.jar:dependencies/lib/main/org.mongodb.mongodb-driver-sync-4.1.0.jar:dependencies/lib/main/org.reactivestreams.reactive-streams-1.0.3.jar:dependencies/lib/main/org.slf4j.slf4j-api-1.7.30.jar:dependencies/quarkus-app-dependencies.txt:dependencies/quarkus-run.jar:dependencies/quarkus/generated-bytecode.jar:dependencies/quarkus/quarkus-application.dat:dependencies/quarkus/transformed-bytecode.jar io.quarkus.bootstrap.runner.QuarkusEntryPoint
2021-04-23 11:24:01,091 INFO  [org.mon.dri.cluster] (main) Cluster created with settings {hosts=[mongodb:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms'}
2021-04-23 11:24:01,110 INFO  [org.mon.dri.cluster] (main) Cluster created with settings {hosts=[mongodb:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms'}
2021-04-23 11:24:01,143 INFO  [org.mon.dri.connection] (cluster-rtt-ClusterId{value='6082ae51fd5b67558bd8a0ef', description='null'}-mongodb:27017) Opened connection [connectionId{localValue:1, serverValue:96}] to mongodb:27017
2021-04-23 11:24:01,143 INFO  [org.mon.dri.connection] (cluster-ClusterId{value='6082ae51fd5b67558bd8a0ef', description='null'}-mongodb:27017) Opened connection [connectionId{localValue:2, serverValue:97}] to mongodb:27017
2021-04-23 11:24:01,147 INFO  [org.mon.dri.connection] (cluster-ClusterId{value='6082ae51fd5b67558bd8a0f0', description='null'}-mongodb:27017) Opened connection [connectionId{localValue:3, serverValue:99}] to mongodb:27017
2021-04-23 11:24:01,147 INFO  [org.mon.dri.connection] (cluster-rtt-ClusterId{value='6082ae51fd5b67558bd8a0f0', description='null'}-mongodb:27017) Opened connection [connectionId{localValue:4, serverValue:98}] to mongodb:27017
2021-04-23 11:24:01,154 INFO  [org.mon.dri.cluster] (cluster-ClusterId{value='6082ae51fd5b67558bd8a0f0', description='null'}-mongodb:27017) Monitor thread successfully connected to server with description ServerDescription{address=mongodb:27017, type=REPLICA_SET_PRIMARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=9, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=3960624, setName='rs0', canonicalAddress=10.116.0.51:27017, hosts=[10.116.0.51:27017], passives=[], arbiters=[], primary='10.116.0.51:27017', tagSet=TagSet{[]}, electionId=7fffffff0000000000000002, setVersion=1, topologyVersion=TopologyVersion{processId=608299e720b9d4afb1f10c1a, counter=6}, lastWriteDate=Fri Apr 23 11:23:56 GMT 2021, lastUpdateTimeNanos=5703437692715}
2021-04-23 11:24:01,156 INFO  [org.mon.dri.cluster] (cluster-ClusterId{value='6082ae51fd5b67558bd8a0ef', description='null'}-mongodb:27017) Monitor thread successfully connected to server with description ServerDescription{address=mongodb:27017, type=REPLICA_SET_PRIMARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=9, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=18244290, setName='rs0', canonicalAddress=10.116.0.51:27017, hosts=[10.116.0.51:27017], passives=[], arbiters=[], primary='10.116.0.51:27017', tagSet=TagSet{[]}, electionId=7fffffff0000000000000002, setVersion=1, topologyVersion=TopologyVersion{processId=608299e720b9d4afb1f10c1a, counter=6}, lastWriteDate=Fri Apr 23 11:23:56 GMT 2021, lastUpdateTimeNanos=5703432262507}
2021-04-23 11:24:01,173 INFO  [org.apa.cam.k.Runtime] (main) Apache Camel K Runtime 1.7.0
2021-04-23 11:24:01,174 INFO  [org.apa.cam.qua.cor.CamelBootstrapRecorder] (main) bootstrap runtime: org.apache.camel.quarkus.main.CamelMainRuntime
2021-04-23 11:24:01,199 INFO  [org.apa.cam.k.lis.SourcesConfigurer] (main) Loading routes from: SourceDefinition{name='MyRouteBuilder', language='java', location='file:/etc/camel/sources/MyRouteBuilder.java', }
2021-04-23 11:24:01,866 INFO  [org.apa.cam.com.azu.sto.que.QueueComponent] (main) No QueueServiceClient instance in the registry
2021-04-23 11:24:01,875 INFO  [org.apa.cam.imp.eng.AutowiredLifecycleStrategy] (main) Autowired property: mongoConnection on component: mongodb as exactly one instance of type: com.mongodb.client.MongoClient (com.mongodb.client.MongoClient_d2b13ff4bbc14069809c540bdc885c49da4586c6_Synthetic_ClientProxy) found in the registry
2021-04-23 11:24:01,896 ERROR [org.apa.cam.qua.mai.CamelMainRuntime] (main) Failed to start application: org.apache.camel.FailedToStartRouteException: Failed to start route route1 because of null
        at org.apache.camel.impl.engine.RouteService.setUp(RouteService.java:132)
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doInitRoutes(InternalRouteStartupManager.java:92)
        at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2773)
        at org.apache.camel.quarkus.core.FastCamelContext.doInit(FastCamelContext.java:546)
        at org.apache.camel.support.service.BaseService.init(BaseService.java:83)
        at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2490)
        at org.apache.camel.support.service.BaseService.start(BaseService.java:111)
        at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2507)
        at org.apache.camel.quarkus.main.CamelMain.doStart(CamelMain.java:94)
        at org.apache.camel.support.service.BaseService.start(BaseService.java:115)
        at org.apache.camel.quarkus.main.CamelMain.startEngine(CamelMain.java:139)
        at org.apache.camel.quarkus.main.CamelMainRuntime.start(CamelMainRuntime.java:49)
        at org.apache.camel.quarkus.core.CamelBootstrapRecorder.start(CamelBootstrapRecorder.java:45)
        at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot-173480958.deploy_0(CamelBootstrapProcessor$boot-173480958.zig:101)
        at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot-173480958.deploy(CamelBootstrapProcessor$boot-173480958.zig:40)
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:807)
        at io.quarkus.runtime.Application.start(Application.java:90)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:100)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:119)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:48)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:25)
Caused by: java.lang.NullPointerException
        at org.apache.camel.component.azure.storage.queue.QueueConsumer.<init>(QueueConsumer.java:53)
        at org.apache.camel.component.azure.storage.queue.QueueEndpoint.createConsumer(QueueEndpoint.java:59)
        at org.apache.camel.impl.engine.DefaultRoute.gatherRootServices(DefaultRoute.java:593)
        at org.apache.camel.impl.engine.DefaultRoute.gatherServices(DefaultRoute.java:577)
        at org.apache.camel.impl.engine.DefaultRoute.initializeServices(DefaultRoute.java:162)
        at org.apache.camel.impl.engine.RouteService.doSetup(RouteService.java:151)
        at org.apache.camel.impl.engine.RouteService.setUp(RouteService.java:130)
        ... 27 more
 
2021-04-23 11:24:01,915 ERROR [io.qua.run.Application] (main) Failed to start application (with profile prod): java.lang.NullPointerException
        at org.apache.camel.component.azure.storage.queue.QueueConsumer.<init>(QueueConsumer.java:53)
        at org.apache.camel.component.azure.storage.queue.QueueEndpoint.createConsumer(QueueEndpoint.java:59)
        at org.apache.camel.impl.engine.DefaultRoute.gatherRootServices(DefaultRoute.java:593)
        at org.apache.camel.impl.engine.DefaultRoute.gatherServices(DefaultRoute.java:577)
        at org.apache.camel.impl.engine.DefaultRoute.initializeServices(DefaultRoute.java:162)
        at org.apache.camel.impl.engine.RouteService.doSetup(RouteService.java:151)
        at org.apache.camel.impl.engine.RouteService.setUp(RouteService.java:130)
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doInitRoutes(InternalRouteStartupManager.java:92)
        at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2773)
        at org.apache.camel.quarkus.core.FastCamelContext.doInit(FastCamelContext.java:546)
        at org.apache.camel.support.service.BaseService.init(BaseService.java:83)
        at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2490)
        at org.apache.camel.support.service.BaseService.start(BaseService.java:111)
        at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2507)
        at org.apache.camel.quarkus.main.CamelMain.doStart(CamelMain.java:94)
        at org.apache.camel.support.service.BaseService.start(BaseService.java:115)
        at org.apache.camel.quarkus.main.CamelMain.startEngine(CamelMain.java:139)
        at org.apache.camel.quarkus.main.CamelMainRuntime.start(CamelMainRuntime.java:49)
        at org.apache.camel.quarkus.core.CamelBootstrapRecorder.start(CamelBootstrapRecorder.java:45)
        at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot-173480958.deploy_0(CamelBootstrapProcessor$boot-173480958.zig:101)
        at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot-173480958.deploy(CamelBootstrapProcessor$boot-173480958.zig:40)
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:807)
        at io.quarkus.runtime.Application.start(Application.java:90)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:100)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:119)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:48)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:25)
```

In camel-k operator logs there is just a periodic reconciliation of the integration:

```
{"level":"info","ts":1619176740.0171115,"logger":"camel-k.controller.integration","msg":"Reconciling Integration","request-namespace":"avano-syndesis","request-name":"azure-queue-to-mongodb"}
{"level":"info","ts":1619176740.0171778,"logger":"camel-k.controller.integration","msg":"Invoking action monitor","request-namespace":"avano-syndesis","request-name":"azure-queue-to-mongodb","api-version":"camel.apache.org/v1","kind":"Integration","ns":"avano-syndesis","name":"azure-queue-to-mongodb"}
{"level":"info","ts":1619176740.0173461,"logger":"camel-k.trait","msg":"Apply trait: init"}
{"level":"info","ts":1619176740.017356,"logger":"camel-k.trait","msg":"Apply trait: camel"}
{"level":"info","ts":1619176740.018646,"logger":"camel-k.trait","msg":"Apply trait: quarkus"}
{"level":"info","ts":1619176740.0186565,"logger":"camel-k.trait","msg":"Apply trait: environment"}
{"level":"info","ts":1619176740.0192816,"logger":"camel-k.trait","msg":"Apply trait: deployer"}
{"level":"info","ts":1619176740.0192928,"logger":"camel-k.trait","msg":"Apply trait: deployment"}
{"level":"info","ts":1619176740.019306,"logger":"camel-k.trait","msg":"Apply trait: gc"}
{"level":"info","ts":1619176740.0198936,"logger":"camel-k.trait","msg":"Apply trait: container"}
{"level":"info","ts":1619176740.0199275,"logger":"camel-k.trait","msg":"Apply trait: jvm"}
{"level":"info","ts":1619176740.0201364,"logger":"camel-k.trait","msg":"Apply trait: owner"}
W0423 11:19:00.097304       1 warnings.go:67] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
{"level":"info","ts":1619177041.0744753,"logger":"camel-k.controller.integration","msg":"Reconciling Integration","request-namespace":"avano-syndesis","request-name":"azure-queue-to-mongodb"}
{"level":"info","ts":1619177041.0745406,"logger":"camel-k.controller.integration","msg":"Invoking action monitor","request-namespace":"avano-syndesis","request-name":"azure-queue-to-mongodb","api-version":"camel.apache.org/v1","kind":"Integration","ns":"avano-syndesis","name":"azure-queue-to-mongodb"}
{"level":"info","ts":1619177041.076645,"logger":"camel-k.trait","msg":"Apply trait: init"}
{"level":"info","ts":1619177041.0766618,"logger":"camel-k.trait","msg":"Apply trait: camel"}
{"level":"info","ts":1619177041.077813,"logger":"camel-k.trait","msg":"Apply trait: quarkus"}
{"level":"info","ts":1619177041.0778253,"logger":"camel-k.trait","msg":"Apply trait: environment"}
{"level":"info","ts":1619177041.0782967,"logger":"camel-k.trait","msg":"Apply trait: deployer"}
{"level":"info","ts":1619177041.0783057,"logger":"camel-k.trait","msg":"Apply trait: deployment"}
{"level":"info","ts":1619177041.0783203,"logger":"camel-k.trait","msg":"Apply trait: gc"}
{"level":"info","ts":1619177041.078861,"logger":"camel-k.trait","msg":"Apply trait: container"}
{"level":"info","ts":1619177041.0788963,"logger":"camel-k.trait","msg":"Apply trait: jvm"}
{"level":"info","ts":1619177041.0790348,"logger":"camel-k.trait","msg":"Apply trait: owner"}
{"level":"info","ts":1619177041.1002555,"logger":"camel-k.controller.integration","msg":"First readiness after 1h26m57s","request-namespace":"avano-syndesis","request-name":"azure-queue-to-mongodb","api-version":"camel.apache.org/v1","kind":"Integration","ns":"avano-syndesis","name":"azure-queue-to-mongodb"}
W0423 11:24:01.339792       1 warnings.go:67] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
{"level":"info","ts":1619177043.0876617,"logger":"camel-k.controller.integration","msg":"Reconciling Integration","request-namespace":"avano-syndesis","request-name":"azure-queue-to-mongodb"}
{"level":"info","ts":1619177043.0877185,"logger":"camel-k.controller.integration","msg":"Invoking action monitor","request-namespace":"avano-syndesis","request-name":"azure-queue-to-mongodb","api-version":"camel.apache.org/v1","kind":"Integration","ns":"avano-syndesis","name":"azure-queue-to-mongodb"}
{"level":"info","ts":1619177043.0902774,"logger":"camel-k.trait","msg":"Apply trait: init"}
{"level":"info","ts":1619177043.0902941,"logger":"camel-k.trait","msg":"Apply trait: camel"}
{"level":"info","ts":1619177043.091416,"logger":"camel-k.trait","msg":"Apply trait: quarkus"}
{"level":"info","ts":1619177043.091428,"logger":"camel-k.trait","msg":"Apply trait: environment"}
{"level":"info","ts":1619177043.0919771,"logger":"camel-k.trait","msg":"Apply trait: deployer"}
{"level":"info","ts":1619177043.0919893,"logger":"camel-k.trait","msg":"Apply trait: deployment"}
{"level":"info","ts":1619177043.092003,"logger":"camel-k.trait","msg":"Apply trait: gc"}
{"level":"info","ts":1619177043.0925844,"logger":"camel-k.trait","msg":"Apply trait: container"}
{"level":"info","ts":1619177043.0926235,"logger":"camel-k.trait","msg":"Apply trait: jvm"}
{"level":"info","ts":1619177043.0927804,"logger":"camel-k.trait","msg":"Apply trait: owner"}
W0423 11:24:03.275351       1 warnings.go:67] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
```

I used following integration, but it is not tied to the components, probably anything that causes camel to fail during startup will reproduce it:

```
package com.test;

import java.lang.Exception;
import java.lang.Override;
import org.apache.camel.builder.RouteBuilder;

public class MyRouteBuilder extends RouteBuilder {
  @Override
  public void configure() throws Exception {
    from("azure-storage-queue://tnb/myqueue?access-key=<accesskey>").to("mongodb://camelMongoClient?database=sampledb&collection=mycollection&operation=insert");
  }
}
```

**Comment by claudio4j** [comment]
I am working on this.

**Comment by davsclaus** [comment]
There is also the NPE in that camel component which would be good to get fixed also

**Comment by oscerd** [comment]
The npe is fixed in 3.10

Il gio 13 mag 2021, 18:36 Claus Ibsen ***@***.***> ha scritto:
